Aptoide/aptoide-client-v8
# [ASV-568] Apps in BN click feedback

**What does this PR do?**

   This aims to improve the interaction with the updates in the BN section. Before, every time a user clicked a button it would take a long time to react. As a temporary solution we decided to set the progress of these apps to indeterminate, until the downloads lib provides us the new download state. This was the solution for every update click.

There is still room for performance improvement.
I have spent a few time researching about the low performance and we think that it is related with Realm and the amount of insertions on the database. Every time there is a progress update or a download status change, there is a database insertion. According to my measurements most of the insertions are taking more than 100ms.

**Database changed?**

 No

**Where should the reviewer start?**

- [ ] AppsManager.java

**How should this be manually tested?**

  Open the apps section in the BN and try to update all apps, or update one individual app. Pause, resume and cancel the update. There should be indeterminate progress to give feedback to the user.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-568](https://aptoide.atlassian.net/browse/ASV-568)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 




**Code Review Checklist**
